bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/GoveeDiscoveryService.java

```diff
@@ -0,0 +1,274 @@
+            final InetSocketAddress socketAddress = new InetSocketAddress(broadcastAddress, DISCOVERY_RESPONSE_PORT);
+
+            getLocalNetworkInterfaces().forEach(localNetworkInterface -> {
+                logger.debug("Discovering Govee Devices on {} ...", localNetworkInterface);
+
+                try (MulticastSocket sendSocket = new MulticastSocket(socketAddress);
+                        MulticastSocket receiveSocket = new MulticastSocket(DISCOVERY_RESPONSE_PORT)) {
+                    sendSocket.setSoTimeout(INTERFACE_TIMEOUT_SEC * MILLIS_PER_SEC);
+                    sendSocket.setReuseAddress(true);
+                    sendSocket.setBroadcast(true);
+                    sendSocket.setTimeToLive(2);
+                    sendSocket.joinGroup(new InetSocketAddress(broadcastAddress, DISCOVERY_RESPONSE_PORT),
+                            localNetworkInterface);
+                    receiveSocket.setReuseAddress(true);
+                    sendBroadcastToDiscoverThing(sendSocket, receiveSocket, broadcastAddress);
+                } catch (IOException e) {
+                    logger.debug("Discovery with IO exception: {}", e.getMessage());
+                }
+            });
+        } catch (InterruptedException ie) {
+            // don't care
+        } catch (UnknownHostException e) {
+            logger.warn("Discovery with UnknownHostException exception: {}", e.getMessage());
+        } finally {
+            discoveryActive = false;
+        }
+    }
+
+    private void sendBroadcastToDiscoverThing(MulticastSocket sendSocket, MulticastSocket receiveSocket,
+            InetAddress broadcastAddress) throws IOException {
+        byte[] requestData = DISCOVER_REQUEST.getBytes();
+
+        DatagramPacket request = new DatagramPacket(requestData, requestData.length, broadcastAddress, DISCOVERY_PORT);
+
+        try {
+            sendSocket.send(request);
+        } catch (SocketTimeoutException ste) {
+            // done with scanning
+        }
+
+        receiveSocket.setSoTimeout(INTERFACE_TIMEOUT_SEC * MILLIS_PER_SEC);
+        receiveSocket.setReuseAddress(true);
+        do {
+            byte[] rxbuf = new byte[10240];
+            DatagramPacket packet = new DatagramPacket(rxbuf, rxbuf.length);
+            receiveSocket.receive(packet);
+
+            String response = new String(packet.getData()).trim();
+            logger.trace("Govee Device Response: {}", response);
+
+            final Map<String, Object> properties = getDeviceProperties(response);
+            final Object product = properties.get(GoveeBindingConstants.PRODUCT_NAME);
+            final String productName = (product != null) ? product.toString() : "unknown";
+            final Object mac = properties.get(GoveeBindingConstants.MAC_ADDRESS);
+            final String macAddress = (mac != null) ? mac.toString() : "unknown";
+
+            ThingUID thingUid = new ThingUID(GoveeBindingConstants.THING_TYPE_LIGHT, macAddress.replace(":", "_"));
```

Comment:
The mac address is not used during communication, so it would still work but agreed probably only for the first device that has no macAdress. I think that is a total edge and most likely will never happen.  Hence let's keep it unless you have a better idea.

Reply:
Creation is not done anymore in case macAddress is not available.